File: labs/G60_Composite/python/complete/simple_position.py
from position import Position, Status
from datetime import datetime
from enum import Enum

class Direction(Enum):
    Long = 0
    Short = 1

Symbol = str

class SimplePosition(Position):
    symbol: Symbol
    quantity: int
    opened_on: datetime
    closed_on: datetime
    direction: Direction
    status: Status

    def getInfo(self) -> str:
        return '[' + self.symbol + ' ' + str(self.quantity) + ' ' + self.direction.name + ']'

    def __init__(self, symbol: Symbol, quantity: int, direction: Direction=Direction.Long):
        self.symbol=symbol
        self.quantity=quantity
        self.direction=direction
        self.opened_on=datetime.utcnow()
        self.status=Status.Open

File: labs/G60_Composite/java/complete/app/src/main/java/CompositeComplete/SimplePosition.java
package CompositeComplete;

public class SimplePosition extends Position{
    private Direction direction;
    private String symbol;
    private int quantity;

    public Direction getDirection() {
        return direction;
    }

    public String getSymbol() {
        return symbol;
    }

    public int getQuantity() {
        return quantity;
    }

    public static SimplePosition Open(String symbol, int quantity,Direction direction) {
        SimplePosition p = new SimplePosition();

        p.direction = direction;        
        p.symbol = symbol;
        p.quantity = quantity;
    
        return p;
    }

    @Override
    public String toString() {
        return "Position [closedOn=" + this.getClosedOn() + ", direction=" + direction + ", openedOn=" + this.getOpenedOn() + ", quantity="
                + quantity + ", symbol=" + symbol + "]";
    }

}
